Makefile

```
@@ -4,7 +4,12 @@
 PROJECT_NAME ?= upjet-provider-template
 PROJECT_REPO ?= github.com/upbound/$(PROJECT_NAME)
 
-export TERRAFORM_VERSION ?= 1.2.1
+export TERRAFORM_VERSION ?= 1.5.7
+
+# Do not allow a version of terraform greater than 1.5.x, due to versions 1.6+ being
+# licensed under BSL, which is not permitted.
+export TERRAFORM_VERSION_CEILING ?= 1.6
```

nit: I think it makes sense not to allow overriding the ceiling from the environment:
```suggestion
export TERRAFORM_VERSION_CEILING := 1.6
```
At least, this will further prevent something like `TERRAFORM_VERSION_CEILING=1.7 TERRAFORM_VERSION=1.6.0  make build`.

If you like this idea and consider this as a valid concern, I will even suggest not defining the ceiling in a variable and instead hardcoding the string literal `1.6` below as we will have also covered make command-line variables, i.e., the case `make TERRAFORM_VERSION_CEILING=1.7 build`.

---

great idea @ulucinar - I removed this from being a variable at all and have hardcoded this value into the makefile.